pkg/koordlet/statesinformer/kubelet_stub.go

```diff
@@ -19,38 +19,81 @@ package statesinformer
 import (
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
-
-	"github.com/koordinator-sh/koordinator/pkg/util"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/client-go/transport"
 )
 
 type KubeletStub interface {
 	GetAllPods() (corev1.PodList, error)
 }
 
 type kubeletStub struct {
-	ipAddr         string
-	httpPort       int
-	timeoutSeconds int
+	addr       string
+	port       int
+	httpClient *http.Client
 }
 
-func NewKubeletStub(ip string, port, timeoutSeconds int) KubeletStub {
+func NewKubeletStub(addr string, port, timeoutSeconds int, token string) (KubeletStub, error) {
+	preTlsConfig := makeTransportConfig(token, true)
+	tlsConfig, err := transport.TLSConfigFor(preTlsConfig)
+	if err != nil {
+		return nil, err
+	}
+	rt := http.DefaultTransport
+	if tlsConfig != nil {
+		// If SSH Tunnel is turned on
+		rt = utilnet.SetOldTransportDefaults(&http.Transport{
+			TLSClientConfig: tlsConfig,
+		})
+	}
+	roundTripper, err := transport.HTTPWrappersForConfig(makeTransportConfig(token, true), rt)
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{
+		Timeout:   time.Duration(timeoutSeconds) * time.Second,
+		Transport: roundTripper,
+	}
 	return &kubeletStub{
-		ipAddr:         ip,
-		httpPort:       port,
-		timeoutSeconds: timeoutSeconds,
+		httpClient: client,
+		addr:       addr,
+		port:       port,
+	}, nil
+}
+
+func makeTransportConfig(token string, insecure bool) *transport.Config {
+	tlsConfig := &transport.Config{
+		BearerToken: token,
+		TLS: transport.TLSConfig{
+			Insecure: true,
+		},
 	}
+	return tlsConfig
 }
 
 func (k *kubeletStub) GetAllPods() (corev1.PodList, error) {
 	podList := corev1.PodList{}
-	result, err := util.DoHTTPGet("pods", k.ipAddr, k.httpPort, k.timeoutSeconds)
```

Comment:
I assume after this PR, `DoHTTPGet` is no longer needed. Feel free to delete `pkg/util/http.go`.